### PR TITLE
[PKG-2857] datasketches 4.1.0  :snowflake:

### PR DESCRIPTION
**Pull request by skupr-anaconda**
**Current package info**

| Channel | Downloads | Version | Platforms | License |
| --- | --- | --- | --- | --- |
| conda-forge | [Conda Downloads](https://anaconda.org/conda-forge/datasketches) | [Conda Version](https://anaconda.org/conda-forge/datasketches) | [Conda Platforms](https://anaconda.org/conda-forge/datasketches) | Conda License |

Changelog: https://github.com/J535D165/recordlinkage/releases
License: https://github.com/apache/datasketches-cpp/blob/4.1.0/LICENSE and https://github.com/apache/datasketches-cpp/blob/4.1.0/NOTICE
Requirements:
- https://github.com/apache/datasketches-cpp/blob/4.1.0/python/CMakeLists.txt
- https://github.com/apache/datasketches-cpp/blob/4.1.0/CMakeLists.txt
- https://github.com/apache/datasketches-cpp/blob/4.1.0/pyproject.toml
- https://github.com/apache/datasketches-cpp/blob/4.1.0/setup.py

Actions:
1. Add `abs.yaml` for the destination channel
2. Remove [conda_build_config.yaml](https://github.com/AnacondaRecipes/datasketches-feedstock/pull/1/files#diff-ff61408cdc05bc9667deeadb55e4aaceb1371972076b6bf6934f9008920f2bd2)
3. Reset the build number to `0`
4. Add `--no-deps --no-build-isolation` to `script`
5. Move `cmake` and `make` to `build`, add `git` for `linux`.
6. Update `host` dependencies
7. Fix `home` url
8. Update `dev_url`
9. Add `doc_url`
10. Add `description`
11. Add `license_family`

### CODE REVIEW

**Comment by JeanChristopheMorinPerso** [comment]
@skupr-anaconda I unblocked this PR. Note that the CMake uses `ExternalProject_Add` to fetch+compile datasketches-cpp. I don't know if it's what we want or not.

File: recipe/meta.yaml [hunk 1]
```diff
@@ -6,12 +6,17 @@ package:
   version: {{ version }}
 
 source:
-  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/datasketches-{{ version }}.tar.gz
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
   sha256: 6a57e4d1e93ccf1035604e3d9181dbe490281673833ce5be477c197b1195c840
+  patches:
+  - patches/remove-generator-on-windows.patch
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml]: We'll probably want to add `patch` as a dependency on Windows.

Comment (skupr-anaconda) [reply recipe/meta.yaml]: I've added it

**Comment by skupr-anaconda** [comment]
> @skupr-anaconda I unblocked this PR. Note that the CMake uses `ExternalProject_Add` to fetch+compile datasketches-cpp. I don't know if it's what we want or not.

Thanks a lot for unblocking! 
About `ExternalProject_Add`. I found a mention of it only in README.md but not in the source code.

**Review by M-Waszkiewicz-Anaconda — APPROVED** [review]
LGTM

**Review by lorepirri — APPROVED** [review]

**Comment by JeanChristopheMorinPerso** [comment]
>About ExternalProject_Add. I found a mention of it only in README.md but not in the source code.

I see. Yeah the sdist on PyPI contains both the cpp and the python code. I'm guessing they have split the repositories in sine the last release, considering there is https://github.com/apache/datasketches-cpp and https://github.com/apache/datasketches-python. https://github.com/apache/datasketches-cpp contains a 4.1.0 tag which has the python stuff in it, but the main branch doesn't have any python folder. Which leads me to think that the next version will use `ExternalProject_Add`.

**Review by JeanChristopheMorinPerso — APPROVED** [review]

**Pull request merged**